### PR TITLE
smooth out-of-body transition + reduced foot sliding

### PR DESCRIPTION
**Pull request by hyperlogic**
Normally, the hand IK targets will follow the hand controllers exactly.  However when your HMD camera gets too far away from your body, we compute the hand IK targets differently, the hands are relative to your avatar's head instead of where your HMD camera is.   This PR attempts to make this transition as smooth as possible by interpolating between the two positions.

Other minor changes:
* The character rig no-longer animates when the body is prevented from moving due to collision.
* There are new developer > avatar debug options which can be used to turn on foot tracing.  This trace can be helpful for detecting foot sliding.
* Tuned interpolation and animation speeds to reduce foot-sliding

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 10c5da572812572dba9e4e5d77ba04488206648e available at: https://gist.github.com/eb0989c03545a4a558425082e299099d<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8657-10c5da5.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8657-10c5da5.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 203ff958492f1562c63f2d5e6edc9484131840a2 available at: https://gist.github.com/eb0989c03545a4a558425082e299099d<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8657-203ff95.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8657-203ff95.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0a72805874b1294c7fe595975fd7c6724b5a4ebd available at: https://gist.github.com/eb0989c03545a4a558425082e299099d<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8657-0a72805.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8657-0a72805.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: effadd38a6d34705f5497c0cb9d54b167f904dbb available at: https://gist.github.com/eb0989c03545a4a558425082e299099d<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8657-effadd3.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8657-effadd3.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: f9a1a21b2ccfaac9157bfa16ea6a684f507f8224 available at: https://gist.github.com/eb0989c03545a4a558425082e299099d<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8657-f9a1a21.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8657-f9a1a21.dmg

**Comment by hyperlogic** [comment]
sixense download failed on linux build, let's try again.  test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: f9a1a21b2ccfaac9157bfa16ea6a684f507f8224 available at: https://gist.github.com/eb0989c03545a4a558425082e299099d<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8657-f9a1a21.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8657-f9a1a21.dmg

File: libraries/physics/src/CharacterController.cpp [hunk 1]
```diff
@@ -195,6 +195,13 @@ bool CharacterController::checkForSupport(btCollisionWorld* collisionWorld) {
     return hasFloor;
 }
 
+void CharacterController::updateAction(btCollisionWorld* collisionWorld, btScalar deltaTime)
+{
```
Comment (AndrewMeadows) [inline libraries/physics/src/CharacterController.cpp]: Bad formatting.  Curly brace should go on the end of the line above it.

**Comment by hifi-gustavo** [comment]
Build console output for commit: c327c5c6cef31293a6068de7e5abcc2c863d4237 available at: https://gist.github.com/eb0989c03545a4a558425082e299099d<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8657-c327c5c.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8657-c327c5c.dmg

**Pull request merged**